src/main/java/seedu/address/model/contact/Name.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.model.contact;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Contact's name in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidName(String)}
+ */
+public class Name {
+
+    public static final String MESSAGE_CONSTRAINTS =
```

Comment:
"Names should only contain alphabets, spaces and hyphens, and it should not be blank."